python/ray/train/callbacks/results_prepocessors/workers.py

```diff
@@ -0,0 +1,60 @@
+from typing import List, Dict, Iterable, Optional, Union
+
+from ray.train.callbacks.results_prepocessors.preprocessor import \
+    ResultsPreprocessor
+
+
+class WorkersResultsPreprocessor(ResultsPreprocessor):
+    """Preprocesses results by worker."""
+
+    def __init__(self,
+                 workers_to_log: Optional[Union[int, List[int]]] = 0) -> None:
+        self._workers_to_log = self._validate_workers_to_log(workers_to_log)
+
+    def preprocess(self, results: List[Dict]) -> List[Dict]:
+        if self._workers_to_log is None:
+            return results
+
+        results = [
+            result for i, result in enumerate(results)
+            if i in self._workers_to_log
+        ]
+
+        return results
+
+    def _validate_workers_to_log(self, workers_to_log) -> Optional[List[int]]:
+        if isinstance(workers_to_log, int):
+            workers_to_log = [workers_to_log]
+
+        if workers_to_log is not None:
+            if not isinstance(workers_to_log, Iterable):
+                raise TypeError("workers_to_log must be an Iterable, got "
+                                f"{type(workers_to_log)}.")
+            if not all(isinstance(worker, int) for worker in workers_to_log):
+                raise TypeError(
+                    "All elements of workers_to_log must be integers.")
+            if len(workers_to_log) < 1:
+                raise ValueError(
+                    "At least one worker must be specified in workers_to_log.")
+        return workers_to_log
+
+
+class SingleWorkerResultsPreprocessor(WorkersResultsPreprocessor):
```

Comment:
I think we can remove this class now that it's been refactored into a preprocessor. Instead of this extra validation, we can just have the callbacks themselves only accept a single worker index to log.